toolkit.lisp

```diff
@@ -65,18 +65,31 @@
                (warn "No documentation for ~(~a~) ~a." type symb)))))
 
 (defmacro define-docs (&body expressions)
-  `(progn
-     ,@(loop for expr in expressions
-             for length = (length expr)
-             for type = (if (< 2 length) (first expr) 'function)
-             for var = (if (< 2 length) (rest (butlast expr)) (butlast expr))
-             for doc = (car (last expr))
-             collect `(setf ,(funcall (documentation-translator type) var) ,doc))))
+  (multiple-value-bind (formatter expressions)
+      (if (eq :formatter (first expressions))
+          (values (second expressions) (cddr expressions))
+          (values 'basic-docstring-formatter expressions))
+    (let ((!formatter (gensym))
```

Comment:
Please use the convention `foog` for gensyms.

Reply:
Unusual. Do you have document with coding style that you are following?